desktop/mode.py

```diff
@@ -50,115 +51,208 @@ def update(self, kind, **info):
+        self.update_prompt_additions()
+
+    @staticmethod
+    def none():
+        return Mode(None)
 
-            elif self.action == "CREATE NEW":
-                text = ("\nCreate new label from {0} in frame {1}?"
-                    "\nSPACE = CREATE IN FRAME {1} AND ALL SUBSEQUENT FRAMES"
-                    "\nS = CREATE IN FRAME {1} ONLY"
-                    "\nESC = CANCEL").format(self.label, self.frame)
 
-            elif self.action == "FLOOD CELL":
-                text = ("\nFlood selected region of {} with new label in frame {}?"
-                    "\n{}").format(self.label, self.frame, answer)
+class Mode2D(Mode):
+    '''
+    don't need any information about frames
+    '''
 
-            elif self.action == "TRIM PIXELS":
-                text = ("\nTrim unconnected pixels away from selected region of label {} in frame {}?"
-                    "\n{}").format(self.label, self.frame, answer)
+    def __init__(self, kind, **info):
+        super().__init__(kind, **info)
+        self.update_prompt()
 
-            elif self.action == "DELETE":
-                text = ("\nDelete label {} in frame {}?"
-                    "\n{}").format(self.label, self.frame, answer)
+    def set_question_multiframe_options(self):
+        if self.kind == "QUESTION":
 
-            elif self.action == "WATERSHED":
-                text = ("\nPerform watershed to split {}?"
-                    "\n{}").format(self.label_1, answer)
+            if self.action == "REPLACE":
+                self.text = ("\nReplace {} with {}?\n"
+                        + self.simple_answer).format(self.label_2, self.label_1)
 
-            elif self.action == "PREDICT":
-                text = ("\nPredict 3D relationships between labels?"
-                    "\nS = PREDICT THIS FRAME FROM PREVIOUS FRAME"
-                    "\nSPACE = PREDICT ALL FRAMES"
-                    "\nESC = CANCEL")
+            elif self.action == "SWAP":
+                self.text = ("\nSwap {} & {}?\n"
+                        + self.simple_answer).format(self.label_1, self.label_2)
+
+            elif self.action == "CREATE NEW":
+                self.text = ("\nCreate new label from label {0}?\n"
+                        + self.simple_answer).format(self.label)
 
             elif self.action == "RELABEL":
-                text = ("\nRelabel annotations?"
-                    "\nSPACE = RELABEL IN ALL FRAMES"
-                    "\nP = PRESERVE 3D INFO WHILE RELABELING"
-                    "\nS = RELABEL THIS FRAME ONLY"
-                    "\nU = UNIQUELY RELABEL EACH LABEL"
-                    "\nESC = CANCEL")
+                self.text = ("\nRelabel annotations?\n"
+                        + self.simple_answer)
 
-        elif self.kind == "PROMPT":
-            if self.action == "FILL HOLE":
-                text = "\nSelect hole to fill in label {}.".format(self.label)
+    @staticmethod
+    def none():
+        return Mode2D(None)
```

Comment:
Is there anyway to bake this into the parent class so we don't have to redefine it in each child class?